subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -2044,6 +2056,42 @@ static int tls_opt_dtls_role_set(struct tls_context *context,
 	return 0;
 }
 
+#if defined(CONFIG_NET_SOCKETS_TLS_CERT_VERIFY_CALLBACK)
+static int tls_opt_cert_verify_callback_set(struct tls_context *context,
+					    const void *optval,
+					    socklen_t optlen)
+{
+	struct tls_cert_verify_cb *cert_verify;
+
+	if (!optval) {
+		return -EINVAL;
+	}
+
+	if (optlen != sizeof(struct tls_cert_verify_cb)) {
+		return -EINVAL;
+	}
+
+	cert_verify = (struct tls_cert_verify_cb *)optval;
+	if (cert_verify->cb == NULL && cert_verify->ctx != NULL) {
```

Comment:
Why do we care what the ctx is here, or was the idea that user can unset the callback by setting both values null (this is not documented if that is the idea)?

Reply:
I just thought that setting context w/o the callback does not make sense. Perhaps we should just forbid the NULL cb pointer?

Reply:
Is user able to unregister the callback somehow, or does it make sense in the first place?

Reply:
I'm not sure actually if mbed TLS allows that, perhaps better to check for NULL pointer after all. I'll send a fix.